libraries/botbuilder-adapter-twitter/src/twitter.ts

```diff
@@ -0,0 +1,196 @@
+                                }
+                                resolve(response);
+                            });
+                        });
+                        responses.push({ id: res.id_str });
+                    } catch (error) {
+                        throw new Error(`Error parsing activity: ${ error.message }.`);
+                    }
+                    break;
+                default:
+                    responses.push({} as ResourceResponse);
+                    console.warn(`Unsupported activity type: '${ activity.type }'.`);
+                    break;
+            }
+        }
+        return responses;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async updateActivity(context: TurnContext, activity: Partial<Activity>): Promise<void> {
+        throw new Error('Method not supported by the Twitter API.');
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async deleteActivity(context: TurnContext, reference: Partial<ConversationReference>): Promise<void> {
+        throw new Error('Method not supported by Twitter API.');
+    }
+
+    public async continueConversation(reference: Partial<ConversationReference>, logic: (context: TurnContext) => Promise<void>): Promise<void> {
+        const request: Partial<Activity> = TurnContext.applyConversationReference(
+            { type: 'event', name: 'continueConversation' },
+            reference,
+            true
+        );
+        const context: TurnContext = this.createContext(request);
+        return this.runMiddleware(context, logic);
+    }
+
+    public async processActivity(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+
+        const message = await retrieveBody(req);
+
+        if (message == null) {
+            res.status(400);
+            res.end();
+        }
+
+        const activity: Partial<Activity> = {
+            id: message.id,
+            timestamp: new Date(),
+            channelId: this.channel,
+            conversation: {
+                id: message.id,
+                isGroup: false,
+                conversationType: null,
+                tenantId: null,
+                name: ''
+            },
+            from: {
+                id: message.user.id,
+                name: message.user.screen_name
+            },
+            recipient: {
+                id: message.in_reply_to_screen_name,
+                name: message.in_reply_to_screen_name
+            },
+            text: message.text,
+            channelData: message,
+            localTimezone: null,
+            callerId: null,
+            serviceUrl: null,
+            listenFor: null,
+            label: message.id,
+            valueType: null,
+            type: null
+        };
+
+        if (activity.type === ActivityTypes.Message) {
+
+            if(message.entities.media != null && message.entities.media.length > 0) {
```

Comment:
Why not `!==` null? I am more a fan of `===` for type and value checking, will have a look if we can enforce that using ESLint. 

Is `message.entities` always present? Or do we need to check that also.

Reply:
Will have to check against `undefined` for media, think.

Reply:
Changed.